kit/Kit.hpp

```diff
@@ -425,6 +430,10 @@ class Document final : public std::enable_shared_from_this<Document>
     ModifiedState _modified;
     bool _isBgSaveProcess;
     bool _isBgSaveDisabled;
+    std::thread _bgSaveWatchdogThread;
```

Comment:
Not a huge fan - can we encapsulate this into its own class and hide all the details there - I'd really prefer not to grow the Document.

Reply:
Sure.